### ISSUE TITLE
Scened feature pack

### ISSUE THREAD

**Issue by Orfelius** [Spring-SpringBoard/SpringBoard-Core#82]
While the current feature pack is quite good it also:
- includes a couple of wrecks from BA for whatever reason
- infringes on Cavedog ip by having Dragon Teeth
- has the same features but rotated
- has some broken features (the last ones)
- has some poorly scalled features (most notably the grass, moshrooms, "ferns" and lilies
- some features have poorly put together hitobox and volume

solution: purge, burn, fix and release the Scened package instead (or just update to Spring Features 1.2). I can provide a fully loaded list of the state on each map feature provided with the pack.

**Comment by gajop** [comment]
The current feature pack has many issues, but I feel this should be reported upstream instead, and only requests for updates or proposals of different feature packs should be mentioned here.
I can't seem to find "Spring Features v1.2" or "Spring Features 1.2" on rapid.

**Comment by Orfelius** [comment]
Uhby "update" I meant "we fix it" :P I will check whether 0 AD has some new trees or rocks that can be jacked.

**Comment by gajop** [comment]
Oh well that would be great. I suggest this gets coordinated with Forb (at
the linked repo) as he's most likely to maintain it.

2015-06-18 10:59 GMT+09:00 Orfelius <notifications@github.com>:

**Comment by gajop** [comment]
Also related: https://github.com/Spring-Helper-Projects/spring-features/issues/3

**Comment by ForbodingAngel** [comment]
Shit like this really pisses me off.

@Orfelius , you should know better.

> includes a couple of wrecks from BA for whatever reason
You might not want to load up FP or Scened connected to BA or some shit *A mod before testing things.

> infringes on Cavedog ip by having Dragon Teeth
Orly? Please, feel free to look in the source and point out exactly where those files might reside (protip, DISCONNECT FP AND SCENED FROM BA FIRST).

> has the same features but rotated
Spring cannot 45 degrees.

> has some broken features (the last ones)
Engine trees: https://github.com/Spring-Helper-Projects/spring-features/blob/master/features/all%20worlds/treedefs.lua

> has some poorly scalled features (most notably the grass, moshrooms, "ferns" and lilies
Those features are INTENTIONALLY scaled that way. Lathan wanted Alice in Wonderland style mushrooms for Black Lake Swamp. In old versions of riverdale I downscaled quite a few for my own purposes. Feel free to make copies, downscale, and submit pull request.

> some features have poorly put together hitbox and volume
Easily fixed in the def. Pull requests welcome. I tend to fix weird hitboxes as I run across them.

@gajop 
SF has been on version 1.6 for a long time now (like 6 months). What's the problem?

**Comment by gajop** [comment]
There are some reasonable concerns though:
1. Bushy plant, cattails, gruss tuffs, snowy trees, tall grass, weeds are really ugly. I'd remove them all. Not all of the rest are great, but these are just too bad.
2. One of the cactuses seems to be missing a model.
3. Gigantic swam trees aren't rendered properly (I'm having them rendered by gl.FeatureShape but they don't appear on map for some reason).
4. Why did you create 20 engine trees? They all look the same (and are really ugly). Doesn't the engine come with one such featuredef already (and if not, maybe just create a single one instead..?)
5. Lots of features with wrong hitbox (size/offset/midpoint) and radius, but I'll have you fix them later using Scened :P (It has a pretty awesome colvol editing UI now, and I'll want some real-use feedback for it)
6. Scened uses SF 1.6, although I only recently updated it (a week ago?)
7. Spring can already rotate units and features in 360 degrees on the Y axis (http://i.imgur.com/Q5FrMs7.jpg). There are some issues atm when you try to do it on the X and Z (stuff can't point down), but I'm hoping to fix that in the engine for 101.0. Anyhow, definitely no reason to have copies of features that are just rotated.

**Comment by gajop** [comment]
Oh and one more thing: things are named horribly. Many features have the same name, e.g. "An Ancient Formation", some have names that aren't human friendly, e.g. "artBirchMediumLo", and some are just wrong: Pine trees are not Palm trees!

**Comment by ForbodingAngel** [comment]
1. To what end? Just because you don't like them doesn't make them not useful. Scened should be categorizing features (there is actually some category templates in some of the defs that smoth started but isn't used atm).

2. Be more specific?

3. They are there, but they are gigantic. Have a look at black lake swamp. They were made in pieces. Some of the bls features could be removed simply because of the fact that they aren't really useful anywhere else.

4. No they don't look all the same. There are 20(22? -> I think JK knows the exact number)) different varieties of generated engine trees, all corresponding with green dots on a featuremap. By having the feature defs in there, they can be easily manipulated to spawn different feature models. For example, load up a map with engine trees in evo. Evo makes them pretty.

5. That is something that will simply take time to do and will probably be done gradually as those specific features get used. Your colvol editor is a MASSIVE step in the right direction to fixing that issue.

6. Ahh, ok.

7. Backwards compatibility. Y rotation is randomly broken in past versions. It kept getting fixed and rebroken (reminds me of transport unload methods -_- /salt)

8. I herd you you liek pull requests :-P This is actually difficult to fix, because when you're looking at the defs, you don't actually know what you're looking at, so if you're going to give them all descriptions, there ends up being a lot of guessing. I assume the palm thing is probably a bad grep.

To recap: Pull requests are gooood.

**Comment by gajop** [comment]
1. http://i.imgur.com/mVxPGEq.jpg - They're just too bad in comparison with the rest. It's best to be conservative and have a minimum quality level that needs to be satisfied. Ideally this minimum level of quality would increase as new features are added. Goal is quality * quantity.
2. defName : cactus1
3. They seem to be rendered with gl.FeatureShape (http://i.imgur.com/1fhnOjU.jpg) but not when I place them on map (http://i.imgur.com/Gx6R6Tc.jpg)
4. All look identical when I place them (without Evo). Not sure how you get them to be different?
8. Also ideally fixable by exploring them in Scened (I'll ask you to test it when I'm done, still working on a lot of things)

**Comment by gajop** [comment]
A couple of other issues:
1) What are the licenses of these files? As far as I know much of 0AD is licensed under CC-BY-SA and that's a concern (especially if not attributed). Everything should have a clear license!
2) Why is this included with Spring Features? https://github.com/Spring-Helper-Projects/spring-features/tree/master/mapconv%20%2B%20tools

**Comment by ForbodingAngel** [comment]
The 0ad stuff has/had attribution as well as much of the rest of it. The
vast majority is unlicensed and just put out there for the taking by
members of the spring community. There was a time when most of us did stuff
just because we wanted to help people out.

That said, I can tell you the licenses of most of it. For example smoth and
lathan are pd explicitly. The vast majority of the rest of it is pd as well.

Feel free to help me make a comprehensive licenses file 📎.

2) because spring features is a tool specifically for mappers and much of
the time the correct ✔ versions are hard to come by, not to mention all the
old tools that exist that most people don't know about (like grout).

Having it there ensures that it never gets confused or lost.

**Comment by gajop** [comment]
1) If it does, it's well hidden. I think it's pretty important since its licenses may affect any maps made using such features.
I'll see what I can do to help, by at least providing a licensing structure.

2) The inclusion of tools is a but surprising still. "Spring Features" - as I saw it - is a asset package and shouldn't contain any tools. Normally I'd suggest you move this to a tool like Feature Placer or Scened but this is just a collection of Windows-specific .exes and libs, not something that's really useful in a Spring mod.

**Comment by ForbodingAngel** [comment]
Various features should be removed at this point (like black lake swamp features), but as the only one here who actually makes maps reasonably often, I would of course want to decide. Perhaps you could make a list of the ones that ought to be shown the door?

Tools have been moved to a different repo.

Is there a way to actually use SceneD at this point?

**Comment by gajop** [comment]
I've completely removed SF and don't plan on reintroducing them (too many issues, unnecessary dependency, potential conflict with game version and an increase in size). Games can just add them on their own.
Maybe someday we'll have feature packs, maybe not.
